Makefile.f4

```diff
@@ -25,8 +28,16 @@ BINARY		 = $(TARGET_FILE_NAME).bin
 
 all:		$(ELF) $(BINARY)
 
-$(ELF):		$(SRCS) $(MAKEFILE_LIST)
-	$(CC) -o $@ $(SRCS) $(FLAGS)
+# Dependencies for .o files
+-include $(DEPS)
+
+# Compile and generate dependency files
+$(OBJS):	$(SRCS)
+	$(CC) -c $(FLAGS) $*.c -o $*.o
+	$(CC) -MM $(FLAGS) $*.c > $*.d
```

Comment:
These two lines should be checked to see if they can be improved to to be more generic. I don't like that it is hard coded that the c-files become o-files.
```
        $(CC) -c $(FLAGS) $*.c -o $*.o
        $(CC) -MM $(FLAGS) $*.c > $*.d
```